### PR TITLE
Assignment nineteen linq operations

### PR DESCRIPTION
**Pull request by Sruthi-Subaraja**
ASSIGNMENT 19 - LINQ CHALLENGES
To gain a deep understanding of LINQ in c#.
TASK 1 - Basic LINQ queries.
TASK 2 - Complex LINQ queries.
TASK 3 - LINQ to objects.
TASK 4 - LINQ to SQL/ entity framework.
TASK 5 - Performance consideration with LINQ.
TASK 6 - Advanced LINQ features.
TASK 7 - Query builder.

### CODE REVIEW

File: src/Linq/TaskManager.cs [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+    /// Performs all the LINQ operations.
+    /// </summary>
+    public class TaskManager
+    {
+        /// <summary>
+        /// Perform task one - Basic LINQ operations.
+        /// </summary>
+        /// <param name="products">List of products.</param>
+        /// <param name="linqOperations">Instance of LinqOperations.</param>
+        public void RunTaskOne(List<Product> products, LinqOperations linqOperations)
```
Comment (Tharun-Sundar) [inline src/Linq/TaskManager.cs:20]: Average of filtered prices in missing

File: src/Linq/LinqOperations.cs [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+        }
+
+        /// <summary>
+        /// Advanced linq queries.
+        /// </summary>
+        /// <param name="products">List of products available.</param>
+        /// <returns>Ienumerable of a tuple that holds string, int, string.</returns>
+        public IEnumerable<Tuple<string, int, string>> PerformComplexLINQ(List<Product> products)
+        {
+            return products.GroupBy(product => product.Category).Select(product => Tuple.Create(product.Key, product.Count(), product.OrderByDescending(x => x.Price).ElementAt(0).ProductName));
```
Comment (Tharun-Sundar) [inline src/Linq/LinqOperations.cs:31]: The variable used inside Select should be named as productGroup

**Review by Tharun-Sundar — APPROVED** [review]

**Review by Tharun-Sundar — CHANGES_REQUESTED** [review]
Add the pending task

File: src/Linq/LinqOperations.cs (hunk 2 above)
Comment (Sruthi-Subaraja) [reply src/Linq/LinqOperations.cs:31]: Okay

File: src/Linq/TaskManager.cs (hunk 1 above)
Comment (Sruthi-Subaraja) [reply src/Linq/TaskManager.cs:20]: I added it now.

**Review by Tharun-Sundar — APPROVED** [review]

**Pull request merged**